src/main/java/seedu/address/model/ingredient/NumUnits.java

```diff
@@ -18,7 +19,7 @@
      */
     public static final String NUMUNITS_VALIDATION_REGEX = "^\\d+$";
 
-    private final String numberOfUnits;
+    private Integer numberOfUnits;
```

Comment:
Why `Integer` and not `int`? This incurs more cost cos of autoboxing.

Reply:
I think `int` is better
```suggestion
    private int numberOfUnits;
```